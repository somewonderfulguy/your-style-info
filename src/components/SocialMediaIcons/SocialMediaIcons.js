import React from 'react'
import {bool, oneOf} from 'prop-types'

import {FacebookIcon, InstagramIcon, VKontakteIcon, YouTubeIcon, TwitterIcon} from 'assets/images'
import styles from './SocialMediaIcons.module.css'

const propTypes = {
  small: bool,
  color: oneOf(['lightgray', 'darkgray'])
}

const defaultProps = {
  small: false,
  color: 'darkgray'
}

const SMALL_SIZES = new Map([
  ['instagram', {width: 18.9, height: 18.9}],
  ['facebook', {width: 17.5, height: 18.2}],
  ['twitter', {width: 20.3, height: 16.8}],
  ['vk', {width: 17.5, height: 17.5}],
  ['youTube', {width: 22.4, height: 15.4}]
])

const COLORS = new Map([
  ['lightgray', '#919191'],
  ['darkgray', '#636363']
])

const SocialMediaIcons = ({small, color}) => {
  const targetBlank = {rel: 'noopener noreferrer', target: '_blank'}

  const getStyles = type => ({
    ...(small ? SMALL_SIZES.get(type) : {}),
    fill: COLORS.get(color)
  })

  return (
    <div className={small ? styles.layoutSmall : styles.layout}>
      <a href="https://instagram.com/" title="Instagram" {...targetBlank}>
        <InstagramIcon style={getStyles('instagram')} />
      </a>
      <a href="https://www.facebook.com/" style={{marginTop: -2}} title="Facebook" {...targetBlank}>
        <FacebookIcon style={getStyles('facebook')} />
      </a>
      <a href="https://twitter.com/" style={{marginTop: -1}} title="Twitter" {...targetBlank}>
        <TwitterIcon style={getStyles('twitter')} />
      </a>
      <a href="https://vk.com/" title="VKontakte" {...targetBlank}>
        <VKontakteIcon style={getStyles('vk')} />
      </a>
      <a href="https://www.youtube.com/" title="YouTube" {...targetBlank}>
        <YouTubeIcon style={getStyles('youTube')} />
      </a>
    </div>
  )
}

SocialMediaIcons.propTypes = propTypes
SocialMediaIcons.defaultProps = defaultProps

export default SocialMediaIcons