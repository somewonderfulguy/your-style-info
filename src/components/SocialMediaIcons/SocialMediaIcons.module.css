.layout {
  display: flex;
  align-items: center;
}

.layout > * {
  margin-right: 10px;
}
.layout > *:last-child {
  margin-right: 0;
}