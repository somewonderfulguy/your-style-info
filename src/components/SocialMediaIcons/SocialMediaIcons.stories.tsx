import React from 'react'

import { ComponentStory, ComponentMeta } from '@storybook/react'

import SocialMediaIcons from './SocialMediaIcons'

const componentMeta: ComponentMeta<typeof SocialMediaIcons> = {
  title: 'SocialMediaIcons',
  component: SocialMediaIcons
}
export default componentMeta

const Template: ComponentStory<typeof SocialMediaIcons> = (args) => (
  <SocialMediaIcons {...args} />
)

export const SocialMediaIconsDefault = Template.bind({})

SocialMediaIconsDefault.args = {}
