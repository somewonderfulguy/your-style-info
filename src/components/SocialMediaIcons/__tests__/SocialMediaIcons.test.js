import React from 'react'
import {render} from '@testing-library/react'
import snapshotDiff from 'snapshot-diff'

import SocialMediaIcons from '..'

jest.mock('assets/images/social-media-svg-icons')

const setup = (color, small) => (
  render(<SocialMediaIcons color={color} small={small} />)
)

test('snapshot difference: 1) default sizes and colour, 2) smaller sizes, lightgray colour', () => {
  expect(
    snapshotDiff(
      setup().asFragment(),
      setup('lightgray', true).asFragment()
    )
  ).toMatchSnapshot()
})