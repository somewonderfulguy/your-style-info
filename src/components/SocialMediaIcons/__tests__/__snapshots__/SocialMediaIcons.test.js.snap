// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot difference: 1) default sizes and colour, 2) smaller sizes, lightgray colour 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,17 +1,17 @@
  <DocumentFragment>
    <div
-     class=\\"layout\\"
+     class=\\"layoutSmall\\"
    >
      <a
        href=\\"https://instagram.com/\\"
        rel=\\"noopener noreferrer\\"
        target=\\"_blank\\"
        title=\\"Instagram\\"
      >
        <svg
-         style=\\"fill: #636363;\\"
+         style=\\"width: 18.9px; height: 18.9px; fill: #919191;\\"
        >
          <path
            d=\\"InstagramIcon\\"
          />
        </svg>
@@ -22,11 +22,11 @@
        style=\\"margin-top: -2px;\\"
        target=\\"_blank\\"
        title=\\"Facebook\\"
      >
        <svg
-         style=\\"fill: #636363;\\"
+         style=\\"width: 17.5px; height: 18.2px; fill: #919191;\\"
        >
          <path
            d=\\"FacebookIcon\\"
          />
        </svg>
@@ -37,11 +37,11 @@
        style=\\"margin-top: -1px;\\"
        target=\\"_blank\\"
        title=\\"Twitter\\"
      >
        <svg
-         style=\\"fill: #636363;\\"
+         style=\\"width: 20.3px; height: 16.8px; fill: #919191;\\"
        >
          <path
            d=\\"TwitterIcon\\"
          />
        </svg>
@@ -51,11 +51,11 @@
        rel=\\"noopener noreferrer\\"
        target=\\"_blank\\"
        title=\\"VKontakte\\"
      >
        <svg
-         style=\\"fill: #636363;\\"
+         style=\\"width: 17.5px; height: 17.5px; fill: #919191;\\"
        >
          <path
            d=\\"VKontakteIcon\\"
          />
        </svg>
@@ -65,11 +65,11 @@
        rel=\\"noopener noreferrer\\"
        target=\\"_blank\\"
        title=\\"YouTube\\"
      >
        <svg
-         style=\\"fill: #636363;\\"
+         style=\\"width: 22.4px; height: 15.4px; fill: #919191;\\"
        >
          <path
            d=\\"YouTubeIcon\\"
          />
        </svg>"
`;
